In [14]:
import pandas as pd

url = "https://drive.google.com/uc?id=1Mlu9r3afSGz5_02Y3wDJoQLpXszQXaIU"
df = pd.read_csv(url,index_col='order_id')
print("First 5 rows")
display(df.head())
print("Last 5 rows")
display(df.tail())

First 5 rows


,customer_name,product,quantity,unit_price,order_date,status
order_id,,,,,,
1001,Alice Johnson,Laptop,1.0,85000.0,2024-01-15,delivered
1002,Bob Smith,Mouse,3.0,499.0,2024-01-16,pending
1003,Carol White,Monitor,2.0,15000.0,NaN,delivered
1004,Dave Brown,Keyboard,NaN,1200.0,2024-01-18,cancelled
1005,Eva Green,Laptop,1.0,85000.0,2024-01-19,delivered


Last 5 rows


,customer_name,product,quantity,unit_price,order_date,status
order_id,,,,,,
1016,Paul Harris,Mouse,6.0,499.0,2024-01-30,pending
1017,Quinn Martin,Monitor,NaN,14500.0,2024-01-31,delivered
1018,Rachel King,Laptop,1.0,85000.0,NaN,delivered
1019,Sam Wright,Keyboard,3.0,1200.0,2024-02-01,cancelled
1020,Tina Baker,USB Hub,2.0,799.0,2024-02-02,pending


In [18]:
display(df.info())
display(df.isnull().sum())
display(df.shape)

df['order_date'] = pd.to_datetime(df['order_date'])
display(df.info())


<class 'pandas.core.frame.DataFrame'>
Index: 20 entries, 1001 to 1020
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   customer_name  20 non-null     object 
 1   product        20 non-null     object 
 2   quantity       16 non-null     float64
 3   unit_price     20 non-null     float64
 4   order_date     16 non-null     object 
 5   status         20 non-null     object 
dtypes: float64(2), object(4)
memory usage: 1.1+ KB


None

,0
customer_name,0
product,0
quantity,4
unit_price,0
order_date,4
status,0


(20, 6)

<class 'pandas.core.frame.DataFrame'>
Index: 20 entries, 1001 to 1020
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   customer_name  20 non-null     object        
 1   product        20 non-null     object        
 2   quantity       16 non-null     float64       
 3   unit_price     20 non-null     float64       
 4   order_date     16 non-null     datetime64[ns]
 5   status         20 non-null     object        
dtypes: datetime64[ns](1), float64(2), object(3)
memory usage: 1.1+ KB


None

In [19]:
df.describe()

,quantity,unit_price,order_date
count,16.000000,20.000000,16
mean,2.437500,20224.600000,2024-01-24 09:00:00
min,1.000000,499.000000,2024-01-15 00:00:00
25%,1.000000,799.000000,2024-01-19 18:00:00
50%,2.000000,2500.000000,2024-01-24 12:00:00
75%,3.000000,14625.000000,2024-01-29 06:00:00
max,6.000000,85000.000000,2024-02-02 00:00:00
std,1.504161,33573.389607,NaN


1)The median unit_price is 2500.00.

2)The unit_price distribution appears to be right-skewed because the mean (20224.60) is significantly higher than the median (2500.00), and the maximum value (85000) is much larger than the 75th percentile (14625), indicating the presence of high-value outliers pulling the distribution to the right.

